src/brasil/gov/agenda/browser/agenda.py

```diff
@@ -37,15 +38,20 @@ def agenda_recente(self):
         """Deve retornar a agendadiaria para o dia atual
            caso contrario exibimos
         """
-        agenda = None
-        hoje = DateTime().strftime(AGENDADIARIAFMT)
-        # Validamos se existe uma agenda para o dia de hoje
-        # e se ela esta publicada
-        if hoje in self.context.objectIds():
-            agenda = self.context[hoje]
-            review_state = self.workflow.getInfoFor(agenda, 'review_state')
-            agenda = agenda if review_state == 'published' else None
-        return agenda
+        brains = api.content.find(
+            context=self.context,
+            object_provides=IAgendaDiaria,
+            effective={
+                'query': (DateTime() - 365, DateTime() + 0.1),
+                'range': 'min:max',
+            },
+            sort_on='effective',
+            sort_order='reverse',
+            sort_limit=1,
+            review_state='published',
+        )
+        if brains:
```

Comment:
what happens if we have no results?

Reply:
a method without ``return`` statement returns ``None`` by default

Reply:
but ``explicit`` is better than ``implicit``.. I'll change the code ;)